### PR TITLE
Add call and arg arguments to `vec_rep()` and friends

### LINKED ISSUE #1303: Add x_arg/times_arg arguments to vec-rep and vec-rep-each
That could have been useful in `uncount()`, where `times` is really `weights`
https://github.com/tidyverse/tidyr/pull/1070#discussion_r537527020

### PR DESCRIPTION
**Pull request by lionel-**
Closes #1303.

### CODE REVIEW

File: R/rep.R [hunk 1]
```diff
@@ -83,14 +85,22 @@ NULL
 
 #' @rdname vec-rep
 #' @export
-vec_rep <- function(x, times) {
-  .Call(ffi_vec_rep, x, times)
+vec_rep <- function(x,
+                    times,
```
Comment (DavisVaughan) [inline R/rep.R:89]: Strongly recommend adding empty `...` in both of these functions to require named optional args

File: R/rep.R [hunk 2]
```diff
@@ -36,6 +36,8 @@
 #'   For `vec_rep_each()`, an integer vector of the number of times to repeat
 #'   each element of `x`. `times` will be [recycled][vector_recycling_rules] to
 #'   the size of `x`.
+#' @inheritParams rlang::args_error_context
```
Comment (DavisVaughan) [inline R/rep.R:39]: I thought we always put `@inheritParams` tags above all other `@param` tags (i.e. `@inheritParams` would be the first tag listed)

File: src/rep.c [hunk 3]
```diff
@@ -1,11 +1,17 @@
+#include "rlang.h"
 #include "vctrs.h"
 #include "type-data-frame.h"
 #include "decl/rep-decl.h"
+#include "vec.h"
```
Comment (DavisVaughan) [inline src/rep.c:5]:
```suggestion
#include "vec.h"
#include "decl/rep-decl.h"
```

I think we always want decl/ headers last?

File: src/rep.c [hunk 4]
```diff
@@ -379,9 +440,4 @@ r_obj* new_unrep_data_frame(r_obj* key, r_obj* times, r_ssize size) {
 
 // -----------------------------------------------------------------------------
 
-void vctrs_init_rep(r_obj* ns) {
-  args_times_ = new_wrapper_arg(NULL, "times");
-}
-
-static
-struct vctrs_arg* const p_args_times = &args_times_;
+void vctrs_init_rep(r_obj* ns) { }
```
Comment (DavisVaughan) [inline src/rep.c:443]: I'm fine with removing this altogether now. We can add it back if we need it

**Review by DavisVaughan — APPROVED** [review]

File: R/rep.R (hunk 1 above)
Comment (lionel-) [reply R/rep.R:89]: oh yeah

File: src/rep.c (hunk 4 above)
Comment (lionel-) [reply src/rep.c:443]: I think it's ok to leave it there, and then it's less work if we need it again. I.e. with ideal tooling every file would get a init hook automatically.

**Pull request merged**